packages/SwingSet/src/kernel/vat-loader/transcript.js

```diff
@@ -59,13 +136,44 @@ export function makeTranscriptManager(
   let replayError;
 
   function simulateSyscall(newSyscall) {
-    const s = playbackSyscalls.shift();
-    const newReplayError = compareSyscalls(vatID, s.d, newSyscall);
-    if (newReplayError) {
-      replayError = newReplayError;
-      throw replayError;
+    while (playbackSyscalls.length) {
+      const compareError = compareSyscalls(
+        vatID,
+        playbackSyscalls[0].d,
+        newSyscall,
+      );
+
+      if (compareError === missingSyscall) {
+        // return `undefined` to indicate that this syscall cannot be simulated
+        // and needs to be performed (virtual collection metadata get)
+        return undefined;
+      }
+
+      const s = playbackSyscalls.shift();
+
+      if (!compareError) {
+        return s.response;
+      } else if (compareError !== extraSyscall) {
+        replayError = compareError;
```

Comment:
I've walked through the cases and convinced myself this will do the right thing, but it'd be lovely to have a short unit test that demonstrates the different permutations this is prepared to handle.

Reply:
Yeah I've thought about it, but given it's not something I plan on maintaining long term, I didn't. Happy to think about what some unit tests would look like if you feel strongly about them here.